bika/lims/catalog/indexers/analysisrequest.py

```diff
@@ -40,12 +40,15 @@ def listing_searchable_text(instance):
     entries = set()
     catalog = api.get_tool(CATALOG_ANALYSIS_REQUEST_LISTING)
     columns = catalog.schema()
+    brains = catalog({"UID": api.get_uid(instance)})
 
     for column in columns:
-        value = api.safe_getattr(instance, column, None)
-        parsed = api.to_searchable_text_metadata(value)
-        if parsed:
-            entries.add(parsed)
+        brain_value = None
+        if brains:
+            brain_value = api.safe_getattr(brains[0], column, None)
```

Comment:
Please put this logic outside of the loop. Retrieve the `brain` once and do a `brain_value = api.safe_getattr(brain, column, None)` here

Reply:
Note that even if the `brain` is `None` at this point, something like this still works:
`brain_value = getattr(None, "foo", None)`